compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -432,7 +433,7 @@ public void generateVarLoad(MethodVisitor mv, BIRNode.BIRVariableDcl varDcl, int
 
         switch (varDcl.kind) {
             case SELF:
-                mv.visitVarInsn(ALOAD, 0);
+                mv.visitVarInsn(ALOAD, this.indexMap.get(OBJECT_SELF_INSTANCE));
```

Comment:
Should this be always constant. Can we make class level variable

Reply:
This can be different from method to method. Since accessing the hashmap is O(1) and adding another variable to JVMInstruction always will always take memory eventhough it is not necessary, will keep this like this.